xds/internal/balancer/clusterresolver/balancerconfig/configbuilder.go

```diff
@@ -205,8 +202,8 @@ func buildClusterImplConfigForEDS(parentPriority int, edsResp xdsclient.Endpoint
 // - ["p0", "p1"]
 // - map{"p0":[L0, L1], "p1":[L2]}
 func groupLocalitiesByPriority(localities []xdsclient.Locality) ([]string, map[string][]xdsclient.Locality) {
-	var priorityIntSlice []int
-	priorities := make(map[string][]xdsclient.Locality)
+	priorities := make(map[string][]xdsclient.Locality, len(localities))
+	priorityIntSlice := make([]int, 0, len(localities))
```

Comment:
Same here. There is a `continue` within the `for` loop. So, we should revert this.